src/components/structures/RoomView.js

```diff
@@ -602,7 +617,7 @@ module.exports = React.createClass({
     },
 
     onRoom: function(room) {
-        if (!room || room.roomId !== this.state.roomId) {
+        if (!room || room.roomId !== this.state.roomId || !this.state.joining) {
```

Comment:
Doesn't this race? ie. the join request could return before the room comes down the sync, in which case we'll never set the room?